test/testcafe/spec/ChallengeOktaVerifyFastPassView_spec.js

```diff
@@ -15,11 +16,14 @@ import identifyWithUserVerificationLaunchUniversalLink from '../../../playground
 import mfaSelect from '../../../playground/mocks/data/idp/idx/authenticator-verification-select-authenticator';
 import loopbackChallengeNotReceived from '../../../playground/mocks/data/idp/idx/identify-with-device-probing-loopback-challenge-not-received';
 import assureWithLaunchAppLink from '../../../playground/mocks/data/idp/idx/authenticator-verification-okta-verify-signed-nonce-app-link';
+import badRequestError from  '../../../playground/mocks/data/idp/idx/error-400-bad-request.json';
 import { renderWidget } from '../framework/shared';
 
 const BEACON_CLASS = 'mfa-okta-verify';
 
 let probeSuccess = false;
+let firstPollCalledForRedundantPolling = false;
+let firstPollCalledForEnhancedPolling = false;
```

Comment:
Instead of using these flags, would it be possible to attempt using testcafe's `removeRequestHooks` and `addRequesthooks` functions? 

You can see an example of how it would work [here](https://medium.com/@kishore_nanduri/testcafe-handling-chaining-mock-requests-8d6cd175d36f). The general idea is that there will be two mocks defined (one for when `firstPollCalledForRedundantPolling` is true and one when it is false) and within the test, you will assert on the intial polling request (i.e. using the logger to check that the poll request happened) then after that is true, you can use `removeRequestHooks` to remove the first mock and then `addRequestHooks` to add the second hook. 

If you need any assistance with this please let me know, we have an example in this repo [here](https://github.com/okta/okta-signin-widget/blob/0d25f438ba0a31e3f4b2327a8b439d2cee6806f8/test/testcafe/spec/EnrollAuthenticatorOktaVerify_spec.js#L43-L51)

Reply:
The reason for recommending this is that we have seen alot of flakiness when using these variables within the `respond` callback.

Reply:
yeah originally i used same var for these two values and noticed the flakiness. then i decided to make them two distinct variables.
I will try to use the way you recommended. but i'm curious know how i can verify there is no flakiness on the test? do we have a similar tests that can run same test for multiple times in parallel which we usually do in monolith?

Reply:
The best way to test against flakiness is by running the test with increased concurrency. 

i.e. 
```
yarn testcafe -c 3 chrome:headless test/testcafe/spec/ChallengeOktaVerifyFastPassView_spec.js
```

where `-c 3` denotes the number of concurrent browsers, increasing that should expose flakiness.

Reply:
since this test use case is tricky, add/remove request mocks won't work properly. I verified parallel run in my local by running `yarn testcafe -c 3 chrome "test/testcafe/spec/ChallengeOktaVerifyFastPassView_spec.js` and didn't spot any flakiness. So will just keep it as it is.